src/leaguepedia_client.py

```
@@ -112,12 +140,86 @@ async def get_scoreboard_data(self, overview_page: str):
         except Exception as e:
+            # Some mwrogue / mediawiki clients sometimes raise structured tuples
+            # or messages that include 'ratelimit'/'ratelimited'. Detect that and
+            # apply an exponential backoff per overview_page.
+            s = repr(e).lower()
+            if "ratelimit" in s or "ratelimited" in s:
+                # brief pause to avoid immediate retry storms from concurrent tasks
+                try:
+                    await asyncio.sleep(5)
+                except Exception:
+                    # Ignore sleep cancellation issues and continue to set backoff
+                    pass
```

Ruff (0.14.10)

`try`-`except`-`pass` detected, consider logging the exception

(S110)